src/hotspot/share/runtime/threadCrashProtection.hpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2017, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+
+#ifndef SHARE_RUNTIME_THREADCRASHPROTECTION_HPP
+#define SHARE_RUNTIME_THREADCRASHPROTECTION_HPP
+
+#include "memory/allocation.hpp"
+
+class CrashProtectionCallback : public StackObj {
+public:
+  virtual void call() = 0;
+};
+
+#if defined(LINUX) || defined(AIX) || defined(BSD)
+# include "threadCrashProtection_posix.hpp"
+#elif defined(_WINDOWS)
+# include "threadCrashProtection_windows.hpp"
+#else
+# error "No ThreadCrashProtection implementation provided for this OS"
+#endif
```

Comment:
Shouldn't you use this?
#define OS_HEADER(basename)            XSTR(OS_HEADER_STEM(basename).hpp)

Reply:
I fixed it as you suggested. I also fixed semaphore.hpp where I copied the old pattern from.

Reply:
Doesn't the OS_HEADER() macro handle the posix case too?

Reply:
` OS_HEADER(threadCrashProtection)` will give threadCrashProtection_linux.hpp, etc. It won't give us threadCrashProtection_posix.hpp. I didn't want to write three threadCrashProtection_{aix,bsd,linux}.hpp files that each just has a single line that includes the _posix version.

Reply:
Yeah no posix support with OS_HEADER.